### PR TITLE
CIV-7994 Integrate with HMC API - Ticket 2

### PR DESCRIPTION
**Pull request by adam-thomas-HMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-7994

### Change description ###

Model the PUT/partiesNotified element for HMC API integration 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/client/HearingsApi.java [hunk 1]
```diff
@@ -35,7 +37,17 @@ HearingGetResponse getHearingRequest(
     PartiesNotifiedResponses getPartiesNotifiedRequest(
         @RequestHeader(AUTHORIZATION) String authorisation,
         @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorization,
-        @PathVariable("id") String id
+        @RequestParam("id") String id
```
Comment (sherlynkhaw) [inline commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/client/HearingsApi.java:40]: not a problem but make sure you rebase before merging with master

it should be 
```@PathVariable("id") String id```

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/model/unnotifiedhearings/PartiesNotified.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.hmcts.reform.hmc.model.unnotifiedhearings;
```
Comment (sherlynkhaw) [inline commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/model/unnotifiedhearings/PartiesNotified.java:1]: this model/file isn't used anywhere?

**Review by sherlynkhaw — CHANGES_REQUESTED** [review]

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/model/unnotifiedhearings/PartiesNotified.java (hunk 2 above)
Comment (adam-thomas-HMCTS) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/model/unnotifiedhearings/PartiesNotified.java]: Removed

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/service/HearingsService.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                hearingId);
+        } catch (FeignException e) {
+            log.error("Failed to retrieve patries notified with Id: %s from HMC", hearingId);
+            throw new HmcException(e);
+        }
+    }
+
+    public ResponseEntity updatePartiesNotifiedResponse(String authToken, String hearingId, int requestVersion, LocalDateTime receivedDateTime) {
+        try {
+            return hearingNoticeApi.updatePartiesNotifiedRequest(
```
Comment (Gareth40343) [inline commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/service/HearingsService.java:53]: pass serviceData here too

**Review by Gareth40343 — CHANGES_REQUESTED** [review]

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/service/HearingsService.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                authToken,
+                authTokenGenerator.generate(),
+                hearingId);
+        } catch (FeignException e) {
+            log.error("Failed to retrieve patries notified with Id: %s from HMC", hearingId);
+            throw new HmcException(e);
+        }
+    }
+
+    public ResponseEntity updatePartiesNotifiedResponse(String authToken, String hearingId, int requestVersion, LocalDateTime receivedDateTime) {
```
[the remark this answers is not in the document]
Comment (adam-thomas-HMCTS) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/service/HearingsService.java]: Changes made

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/client/HearingsApi.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    @GetMapping(value = PARTIES_NOTIFIED_ENDPOINT, consumes = MediaType.APPLICATION_JSON_VALUE)
+    PartiesNotifiedResponses getPartiesNotifiedRequest(
+        @RequestHeader(AUTHORIZATION) String authorisation,
+        @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorization,
+        @PathVariable("id") String id
+    );
+
+    @PutMapping(value = PARTIES_NOTIFIED_ENDPOINT)
+    ResponseEntity updatePartiesNotifiedRequest(
+        @RequestHeader(AUTHORIZATION) String authorisation,
```
[the remark this answers is not in the document]
Comment (adam-thomas-HMCTS) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/client/HearingsApi.java:47]: Changes made

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/service/HearingsService.java (hunk 3 above)
Comment (adam-thomas-HMCTS) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/service/HearingsService.java:56]: changes made

**Review by Gareth40343 — APPROVED** [review]

**Review by sankaviv1 — DISMISSED** [review]

File: commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/model/unnotifiedhearings/PartiesNotified.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
+public class PartiesNotified {
+
+    private Integer requestVersion;
```
Comment (sherlynkhaw) [inline commons-civil/src/main/java/uk/gov/hmcts/reform/hmc/model/unnotifiedhearings/PartiesNotified.java:15]: you have request version in this and another in the api, it's duplicated?

**Review by sherlynkhaw — CHANGES_REQUESTED** [review]

**Review by sherlynkhaw — APPROVED** [review]

**Review by douglasrice — APPROVED** [review]

**Pull request merged**